src/test/suite/didactUriCompletionItemProvider.test.ts

```diff
@@ -0,0 +1,197 @@
+		if (match) {
+			expect(match[0]).to.be.equal('?commandId=vscode.didact.');
+			expect(match[1]).to.be.equal('vscode.didact.');
+		}
+	});
+
+	test("that the match utility returns expected results for didact uri with full command and properties", () => {
+		const match = provider.findMatchForCommandVariable('didact://?commandId=vscode.didact.closeNamedTerminal&text=NamedTerminal');
+		expect(match).to.not.be.null;
+		if (match) {
+			expect(match[0]).to.be.equal('?commandId=vscode.didact.closeNamedTerminal');
+			expect(match[1]).to.be.equal('vscode.didact.closeNamedTerminal');
+		}
+	});
+
+	test("that the command processing for a command prefix returns expected results", async () => {
+		const match = provider.findMatchForCommandVariable('didact://?commandId=vscode.didact.');
+		const completionList = await provider.processCommands(match);
+		expect(completionList.items.length).to.be.equal(26);
+	});
+
+	test("that the command processing for one command returns one expected result", async () => {
+		const match = provider.findMatchForCommandVariable('didact://?commandId=vscode.didact.cliCommandSuccessful&text=cli-requirement-name$$echo%20text');
+		const completionList = await provider.processCommands(match);
+		expect(completionList.items).to.have.lengthOf(1);
+
+		const includeText:string | SnippetString | undefined = completionList.items[0].insertText;
+		expect(includeText).to.not.be.undefined;
+		expect((includeText as SnippetString).value).to.include('${1:Requirement-Label}');
+		expect((includeText as SnippetString).value).to.include('${2:URLEncoded-Command-to-Execute}');
+	});
+
+	test("that the didact protocol matcher returns some expected results for valid values", () => {
+		const validValues : Array<string> = [
+			"(didact://?commandId=mycommand)",
+			"(didact)",
+			"(didact://)",
+			"(didact:)",
+			"(didact:/",
+			"(didact://?",
+			"[my link](didact://?comma)",
+			"link:didact://?commandId=someCommand",
+			"link:didact",
+			"(didact://?commandId=vscode.didact.cliCommandSuccessful&text=cli-requirement-name$$echo%20text)"
+		];
+		suite('testing didact protocol matching against positive results', () => {
+			validValues.forEach(function(value:string){
+				test(`matched a didact protocol in ${value}`, () => {
+					const match = provider.findMatchForDidactPrefix(value);
+					expect(match).to.not.be.null;
+					expect(match?.length).to.be.equal(1);
+				});
+			});
+		});
+	});
+
+	test("that the didact protocol matcher returns some expected results for invalid values", () => {
+		const invalidValues : Array<string> = [
+			"[dooby](did://?",
+			"The man was a didact whiz"
+		];
+		suite('testing didact protocol matching against negative results', () => {
+			invalidValues.forEach(function(value:string){
+				test(`did not match a didact protocol in ${value}`, () => {
+					const match = provider.findMatchForDidactPrefix(value);
+					expect(match).to.be.null;
+					expect(match?.length).to.be.undefined;
```

Comment:
maybe something `expect(match).that.is.empty` would provide a better error message and simplier to read (checking that the length is undefined means that the array is empty right?)

Reply:
`expect(match).that.is.empty;` and `expect(match?.length).to.be.undefined;` are apparently not processed the same. 

I get `AssertionError: .empty was passed non-string primitive null`

But if the match is null, then I shouldn't need to check for an undefined array length, so I will get rid of that line entirely